plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3342,4 +3348,59 @@ public void testDroppingIcebergAndCreatingANewTableWithTheSameNameShouldBePossib
         assertThat(query("SELECT min(a_varchar) FROM test_iceberg_recreate")).matches("VALUES CAST('Trino' AS varchar)");
         dropTable("test_iceberg_recreate");
     }
+
+    @Test
+    public void testPathHiddenColumn()
+    {
+        String tableName = "test_path_" + randomTableSuffix();
+        @Language("SQL") String createTable = "CREATE TABLE " + tableName + " " +
+                "WITH ( partitioning = ARRAY['col1'] ) AS " +
+                "SELECT * FROM (VALUES " +
+                "(0, 0), (3, 0), (6, 0), " +
+                "(1, 1), (4, 1), (7, 1), " +
+                "(2, 2), (5, 2) " +
+                " ) t(col0, col1) ";
+        assertUpdate(createTable, 8);
+        assertTrue(getQueryRunner().tableExists(getSession(), tableName));
+
+        TableMetadata tableMetadata = getTableMetadata(ICEBERG_CATALOG, TPCH_SCHEMA, tableName);
+
+        List<String> columnNames = ImmutableList.of("col0", "col1", PATH_COLUMN_NAME);
+        List<ColumnMetadata> columnMetadatas = tableMetadata.getColumns();
+        assertEquals(columnMetadatas.size(), columnNames.size());
+        for (int i = 0; i < columnMetadatas.size(); i++) {
+            ColumnMetadata columnMetadata = columnMetadatas.get(i);
+            assertEquals(columnMetadata.getName(), columnNames.get(i));
+            if (columnMetadata.getName().equals(PATH_COLUMN_NAME)) {
+                // $path should be hidden column
+                assertTrue(columnMetadata.isHidden());
+            }
+        }
+        int numberOfPartitions = computeActual(format("SELECT partition FROM \"%s$partitions\"", tableName)).getMaterializedRows().size();
+        assertEquals(numberOfPartitions, 3);
+
+        MaterializedResult results = computeActual(format("SELECT *, \"%s\" FROM %s", FILE_PATH.colName(), tableName));
```

Comment:
inline `$path` in the query string.

this must never change, so the test can inline the column name for readability

Reply:
modified this method